athena-cloudera-hive/src/main/java/com/amazonaws/athena/connectors/cloudera/HiveQueryStringBuilder.java

```diff
@@ -28,9 +28,9 @@
 
 public class HiveQueryStringBuilder extends JdbcSplitQueryBuilder
 {
-    public HiveQueryStringBuilder(String quoteCharacters)
+    public HiveQueryStringBuilder(String quoteCharacters, HiveFederationExpressionParser hiveFederationExpressionParser)
```

Comment:
Related to my other comment, if we only made the `HiveFederationExpressionParser` so that we could satisfy the super arguments, we can make the `JdbcFederationExpressionParser` class concrete (remove the abstract modifier) so that it can be instantiated, and change the method `writeArrayConstructorClause` throw a NotImplementedException (as opposed to it being abstract).